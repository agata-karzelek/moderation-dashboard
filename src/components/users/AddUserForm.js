import React from "react";
import { useStyles } from "../../style/createModerationModal";
import {
  DialogContent,
  DialogActions,
  Button,
  Typography,
  TextField,
  CircularProgress,
} from "@material-ui/core";

export default function AddUserForm(props) {
  const classes = useStyles();
  return (
    <form onSubmit={props.formik.handleSubmit}>
      <DialogContent>
        <Typography testid="User_Id" className={classes.subHeadingFont}>
          User ID
        </Typography>
        <TextField
          id="user_id"
          name="user_id"
          placeholder="Enter user ID"
          variant="outlined"
          disabled={props.disabled}
          className={classes.inputBox}
          autoComplete="off"
          onChange={props.formik.handleChange}
          value={props.formik.values.user_id}
          error={props.formik.touched.user_id && Boolean(props.formik.errors.user_id)}
          helperText={props.formik.touched.user_id && props.formik.errors.user_id}
        />
        <Typography testid="User_Name" className={classes.subHeadingFont}>
          User Name
        </Typography>
        <TextField
          id="name"
          name="name"
          placeholder="Enter user name"
          variant="outlined"
          className={classes.inputBox}
          autoComplete="off"
          onChange={props.formik.handleChange}
          value={props.formik.values.name}
          error={props.formik.touched.name && Boolean(props.formik.errors.name)}
          helperText={props.formik.touched.name && props.formik.errors.name}
        />
        <Typography testid="Email" className={classes.subHeadingFont}>
          Email Address
        </Typography>
        <TextField
          id="email"
          name="email"
          placeholder="Enter email address"
          variant="outlined"
          className={classes.inputBox}
          autoComplete="off"
          onChange={props.formik.handleChange}
          value={props.formik.values.email}
          error={props.formik.touched.email && Boolean(props.formik.errors.email)}
          helperText={props.formik.touched.email && props.formik.errors.email}
        />
        <Typography testid="Profile_Url" className={classes.subHeadingFont}>
          Profile URL
        </Typography>
        <TextField
          id="profile_url"
          name="profile_url"
          placeholder="Enter Profile URL"
          variant="outlined"
          className={classes.inputBox}
          autoComplete="off"
          onChange={props.formik.handleChange}
          value={props.formik.values.profile_url}
          error={props.formik.touched.profile_url && Boolean(props.formik.errors.profile_url)}
          helperText={props.formik.touched.profile_url && props.formik.errors.profile_url}
        />
        <br />
      </DialogContent>
      <DialogActions>
        <Button autoFocus className={classes.cancelButton} onClick={props.handleClose}>
          CANCEL
        </Button>

        <Button
          autoFocus
          className={classes.addButton}
          type="submit"
          color="primary"
          disabled={props.userAlert.loading}
          startIcon={
            props.userAlert.loading ? (
              <CircularProgress className={classes.loader} size={30} />
            ) : null
          }
        >
          <Typography className={classes.buttonText}>{props.buttonTitle}</Typography>
        </Button>
      </DialogActions>
    </form>
  );
}
