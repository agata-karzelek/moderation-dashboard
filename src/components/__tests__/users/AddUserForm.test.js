import React from "react";
import { shallow } from "enzyme";
import AddUserForm from "../../users/AddUserForm";

const props = {
  formik: {
    handleSubmit: jest.fn(),
    disabled: false,
    values: {
      user_id: 128,
      name: "test name",
      email: "email@example.com",
      profile_url: "http://example.com",
    },
    touched: {
      user_id: true,
      name: true,
      email: true,
      profile_url: true,
    },
    errors: {
      user_id: true,
      name: true,
      email: true,
      profile_url: true,
    },
  },
  userAlert: {
    loading: true,
  },
};

describe("Add User Test Cases", () => {
  test("Snapshot", () => {
    const wrapper = shallow(<AddUserForm formik={props.formik} userAlert={props.userAlert} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Labels and initial Values", () => {
    const wrapper = shallow(<AddUserForm formik={props.formik} userAlert={props.userAlert} />);
    expect(wrapper.find("[testid='User_Id']").text()).toBe("User ID");
    expect(wrapper.find("[testid='User_Name']").text()).toBe("User Name");
    expect(wrapper.find("[testid='Email']").text()).toBe("Email Address");
    expect(wrapper.find("[testid='Profile_Url']").text()).toBe("Profile URL");
    expect(wrapper.find("[id='user_id']").text()).toBe("");
    expect(wrapper.find("[id='name']").text()).toBe("");
    expect(wrapper.find("[id='email']").text()).toBe("");
    expect(wrapper.find("[id='profile_url']").text()).toBe("");
    expect(wrapper.find("[id='user_id']").getElement().props.placeholder).toBe("Enter user ID");
    expect(wrapper.find("[id='name']").getElement().props.placeholder).toBe("Enter user name");
    expect(wrapper.find("[id='email']").getElement().props.placeholder).toBe("Enter email address");
    expect(wrapper.find("[id='profile_url']").getElement().props.placeholder).toBe(
      "Enter Profile URL"
    );
  });
});
