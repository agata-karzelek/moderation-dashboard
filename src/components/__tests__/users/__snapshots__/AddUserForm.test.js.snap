// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add User Test Cases Snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-subHeadingFont-12"
      testid="User_Id"
    >
      User ID
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-inputBox-6"
      error={true}
      helperText={true}
      id="user_id"
      name="user_id"
      placeholder="Enter user ID"
      value={128}
      variant="outlined"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-subHeadingFont-12"
      testid="User_Name"
    >
      User Name
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-inputBox-6"
      error={true}
      helperText={true}
      id="name"
      name="name"
      placeholder="Enter user name"
      value="test name"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-subHeadingFont-12"
      testid="Email"
    >
      Email Address
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-inputBox-6"
      error={true}
      helperText={true}
      id="email"
      name="email"
      placeholder="Enter email address"
      value="email@example.com"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-subHeadingFont-12"
      testid="Profile_Url"
    >
      Profile URL
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-inputBox-6"
      error={true}
      helperText={true}
      id="profile_url"
      name="profile_url"
      placeholder="Enter Profile URL"
      value="http://example.com"
      variant="outlined"
    />
    <br />
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      className="makeStyles-cancelButton-9"
    >
      CANCEL
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      className="makeStyles-addButton-14"
      color="primary"
      disabled={true}
      startIcon={
        <WithStyles(ForwardRef(CircularProgress))
          className="makeStyles-loader-7"
          size={30}
        />
      }
      type="submit"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-buttonText-5"
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</form>
`;
